File: leetcode/217-contains-duplicate/solution.cpp
#include <bits/stdc++.h>
using namespace std;

// LeetCode 217: Contains Duplicate
class Solution {
public:
    bool containsDuplicate(vector<int>& nums) {
        unordered_set<int> seen;
        for (int num : nums) {
            if (seen.count(num)) return true;
            seen.insert(num);
        }
        return false;
    }
};

int main() {
    Solution sol;
    vector<int> nums = {1, 2, 3, 1};
    cout << boolalpha << sol.containsDuplicate(nums) << endl; // true
    return 0;
}

File: leetcode/217-contains-duplicate/solution.py
from typing import List


# LeetCode 217: Contains Duplicate
class Solution:
    def containsDuplicate(self, nums: List[int]) -> bool:
        return len(set(nums)) < len(nums)


if __name__ == "__main__":
    print(Solution().containsDuplicate([1, 2, 3, 1]))  # True
